### PR TITLE
`pytest`: discover `openpilot/`

### PR DESCRIPTION
**Pull request by BBBmau**
continuation of https://github.com/commaai/openpilot/pull/32489

The idea is to run pytests in just `openpilot` directory. The `discover` part is to go into `openpilot` directory and have all tests passing.

### CODE REVIEW

**Comment by BBBmau** [comment]
not really sure what caused unit tests to be skipped. unit tests passing is the last step for getting this ready. 

Issue is with coverage and how it's trying to handle generated py files.

```
INTERNALERROR> CoverageWarning: Couldn't parse '/tmp/openpilot/config-3.py': No source for code: '/tmp/openpilot/config-3.py'. (couldnt-parse)
```

everywhere else that `$PYTEST` is called passes, so what's left is to fix this error in order to take out of draft.

**Comment by BBBmau** [comment]
@adeebshihadeh only pinging due to unit test being skipped. Most likely still needs work.

**Comment by BBBmau** [comment]
latest commit prevents pytest to run `./selfdrive/ui/tests/test_translations`, this is fine since it's explicitly ran in `selfdrive_tests`

https://github.com/commaai/openpilot/blob/63f55f4915ebb3763cc6bc9adaba52a77467ff23/.github/workflows/selfdrive_tests.yaml#L145

**Comment by BBBmau** [comment]
added back in the `process_replay` after needing to understand why a skip was necessary. makes sense now and the PR can be marked for review since this was the last blocker.

File: .github/workflows/selfdrive_tests.yaml [hunk 1]
```diff
@@ -53,7 +53,7 @@ jobs:
       run: |
         cd $STRIPPED_DIR
         ${{ env.RUN }} "release/check-dirty.sh && \
-                        MAX_EXAMPLES=5 $PYTEST -m 'not slow' selfdrive/car"
+                        MAX_EXAMPLES=5 $PYTEST -m 'not slow' openpilot/selfdrive/car"
```
Comment (adeebshihadeh) [inline .github/workflows/selfdrive_tests.yaml:56]: hmm, do we need to do this for all invocations now? if so, might be better to switch to a list of selfdrive, system, tools, and common instead of openpilot/

File: selfdrive/test/process_replay/test_processes.py [hunk 2]
```diff
@@ -17,6 +17,7 @@
 from openpilot.tools.lib.logreader import LogReader
 from openpilot.tools.lib.helpers import save_log
 
+
```
Comment (adeebshihadeh) [inline selfdrive/test/process_replay/test_processes.py:20]: revert the changes to this file

File: selfdrive/ui/tests/create_test_translations.sh [hunk 3]
```diff
@@ -12,7 +12,6 @@ UNFINISHED="<translation type=\"unfinished\"><\/translation>"
 TRANSLATED="<translation>$TEST_TEXT<\/translation>"
 
 mkdir -p $UI_DIR/translations
-rm -f $TEST_TS_FILE $TEST_QM_FILE
```
Comment (adeebshihadeh) [inline selfdrive/ui/tests/create_test_translations.sh:15]: why was this removed?

Comment (BBBmau) [reply selfdrive/ui/tests/create_test_translations.sh:15]: this can actually be put back in since the issues with the missing `main_test_en` file was from running `pytest test_translations`

Though we could also leave it as is since `lupdate -recursive "$UI_DIR" -ts $TEST_TS_FILE` both generates a `.ts` file and updates `$TEST_TS_FILE` if it already exists.

File: .github/workflows/selfdrive_tests.yaml (hunk 1 above)
Comment (BBBmau) [reply .github/workflows/selfdrive_tests.yaml:56]: we might be able to use `--rootdir` so that we can set the root to be `openpilot`. Next commit will try this.

**Comment by BBBmau** [comment]
nice! setting `--rootdir` prevented any changes to jobs in `selfdrive_test.yaml`

I completely removed testpaths from pyproject since we use `--rootdir` in the yaml test. The testpath would only be used if no directory was set when running `pytest`. We don't need this since `pytest` by itself runs from `openpilot` directory with the new `--rootdir=openpilot` being part of `$PYTEST`

This is ready to be merged, though should we just remove `cpp_files = "test_*.cc"` from `pyproject.toml` also? I don't see how this helps with pytest.

File: selfdrive/ui/tests/create_test_translations.sh (hunk 3 above)
Comment (adeebshihadeh) [reply selfdrive/ui/tests/create_test_translations.sh:15]: the idea is that you have a fresh state every time

**Comment by adeebshihadeh** [comment]
We run our CPP tests with pytest using pytest catch2

File: selfdrive/ui/tests/create_test_translations.sh (hunk 3 above)
Comment (BBBmau) [reply selfdrive/ui/tests/create_test_translations.sh]: makes sense, i added it back in.

File: poetry.lock
Comment (adeebshihadeh) [inline poetry.lock:1]: this shouldn't have been changed

File: .github/workflows/selfdrive_tests.yaml [hunk 4]
```diff
@@ -21,7 +21,7 @@ env:
 
   RUN: docker run --shm-size 2G -v $PWD:/tmp/openpilot -w /tmp/openpilot -e CI=1 -e PRE_COMMIT_HOME=/tmp/pre-commit -e PYTHONWARNINGS=error -e FILEREADER_CACHE=1 -e PYTHONPATH=/tmp/openpilot -e NUM_JOBS -e JOB_ID -e GITHUB_ACTION -e GITHUB_REF -e GITHUB_HEAD_REF -e GITHUB_SHA -e GITHUB_REPOSITORY -e GITHUB_RUN_ID -v $GITHUB_WORKSPACE/.ci_cache/pre-commit:/tmp/pre-commit -v $GITHUB_WORKSPACE/.ci_cache/scons_cache:/tmp/scons_cache -v $GITHUB_WORKSPACE/.ci_cache/comma_download_cache:/tmp/comma_download_cache -v $GITHUB_WORKSPACE/.ci_cache/openpilot_cache:/tmp/openpilot_cache $BASE_IMAGE /bin/bash -c
 
-  PYTEST: pytest --rootdir=openpilot --continue-on-collection-errors --cov --cov-report=xml --cov-append --durations=0 --durations-min=5 --hypothesis-seed 0 -n logical
```
Comment (BBBmau) [inline .github/workflows/selfdrive_tests.yaml]: why was `--rootdir=openpilot` removed? wasn't the goal to discover and run unit tests inside just the openpilot directory? 

Tests are passing since it's just using the directories found in root and not in `/openpilot`

Comment (adeebshihadeh) [reply .github/workflows/selfdrive_tests.yaml]: it changes the local workflow, so i'd have to add "--rootdir" when running locally. this achieves mostly the same thing (discover all tests), but has no effect on the workflow. 

eventually, I think we'll end up moving `selfdrive/` and friends to actually be inside `openpilot/`, and this will all just work.

Comment (BBBmau) [reply .github/workflows/selfdrive_tests.yaml]: ah gotcha, yeah it's a shame `--rootdir` doesn't work when setting it in pyproject.toml
